icu4c/source/test/intltest/loctest.cpp

```diff
@@ -4101,6 +4101,34 @@ LocaleTest::TestCreateUnicodeKeywordSet(void) {
                result.find("ca") != result.end());
     assertTrue("set::find(\"co\")",
                result.find("co") != result.end());
+
+    static const char tag[] = "en-US-u-ca-gregory-x-foo";
+    Locale l2 = Locale::forLanguageTag(tag, status);
+    result.clear();
+    l2.getKeywords<std::string>(
```

Comment:
The way these test cases are structured, it would be more logical if this additional testing of `getKeywords()` was moved into the `TestCreateKeywordSet` test case above, so that it's kept together with the already existing testing of `getKeywords()` there.

Even more logical would be to add new separate test cases, instead of adding new code to the existing ones, so that each one of these test cases continues to simply test one single thing each (that'll make the code simpler, no calls to `clear()`, no variables named `l2`, etc., and a test failure would then point directly to the problem).

Then you'd end up with three test cases for old style keywords:

`TestCreateKeywordSet`
`TestCreateKeywordSetEmpty`
`TestCreateKeywordSetWithPrivateUse`

And three test cases for Unicode keywords:

`TestCreateUnicodeKeywordSet`
`TestCreateUnicodeKeywordSetEmpty`
`TestCreateUnicodeKeywordSetWithPrivateUse`

Reply:
fixed